module-system/sov-modules-stf-template/src/app_template.rs

```diff
@@ -76,10 +82,42 @@ where
             current_storage: storage,
             checkpoint: None,
             phantom_vm: PhantomData,
+            phantom_cond: PhantomData,
             phantom_blob: PhantomData,
         }
     }
 
+    #[cfg(test)]
+    /// Builds a working set from an app template instance. Tries to get the working set from
+    /// the state checkpoint before reaching out to the current storage if the state checkpoint is
+    /// none.
+    pub fn get_working_set(self) -> WorkingSet<C::Storage> {
+        match self.checkpoint {
+            Some(checkpoint) => checkpoint.to_revertable(),
+            None => WorkingSet::new(self.current_storage),
+        }
+    }
+
+    /// Gets a key from the storage, updates the current storage checkpoint to cache the key if not cached yet.
+    pub fn get_from_storage(&mut self, key: StorageKey) -> Option<StorageValue> {
+        match &mut self.checkpoint {
+            Some(ref mut checkpoint) => checkpoint.get(key),
+            None => self.current_storage.get(key, &Default::default()),
+        }
+    }
+
+    /// Helper function that returns a storage value from a key and a prefix.
+    /// It deserializes the storage value to the output type.
+    pub fn get_from_storage_with_prefix<K: BorshSerialize, Out: BorshDeserialize>(
```

Comment:
`get_from_storage_with_prefix` & `get_from_storage` forces a caller to think about prefixses and keys that are internal to modules which we want to avoid. I pointed out in the other comment how these methods can be avoided and I am happy to jump on a call.

Reply:
Ok makes sense, will change that part

Reply:
Ok I got it. There are several ways to handle it. 
1. `AppTemplate` could return a cut reference to the underlying `working_set` but this is not great as it leaks the WS.
2. In tests we can create a new WS based on the DB handle and get all the changes which ware saved there. This is the preferred way.